src/databases/VTK/VTKCommonPluginInfo.C

```diff
@@ -58,11 +52,79 @@ VTKCommonPluginInfo::GetDatabaseType()
 //    I changed the code so it switches the interface based on how many domains
 //    are present in the 1st file.
 //
-//    Kathleen Biagas, Fri Augus 13, 2021
+//    Kathleen Biagas, Fri August 13, 2021
 //    Add MTMD for .pvd file types.
 //
+//    Kathleen Biagas, Thu Dec 12, 2024 
+//    Added tests to see if .pvd is GEOS flavor.
+//
 // ****************************************************************************
 
+#include <FileFunctions.h>
+#include <vtkNew.h>
+#include <vtkXMLDataElement.h>
+#include <vtkXMLDataParser.h>
+using std::string;
+
+bool
+FindGEOSNestedElement(vtkXMLDataElement *el)
+{
+    if(el->GetNumberOfNestedElements() == 0)
+        return false;
+
+    // check for Blocks with certain name attributes used by GEOS
+    if(el->FindNestedElementWithNameAndAttribute("Block", "name","CellElementRegion") ||
+       el->FindNestedElementWithNameAndAttribute("Block", "name","SurfaceElementRegion") ||
+       el->FindNestedElementWithNameAndAttribute("Block", "name","ParticleRegion") ||
+       el->FindNestedElementWithNameAndAttribute("Block", "name","WellElementRegion") )
+    {
+        return true;
+    }
+    for(int i = 0; i < el->GetNumberOfNestedElements(); ++i)
+        return FindGEOSNestedElement(el->GetNestedElement(i));
+}
+
+
+bool
+CheckIfGEOS(const string &rootFile)
```

Comment:
So, I haven't seen actual file I/O being done in the `XXXCommonPluginInfo.C` routines before. I am not saying that is bad. Its just different than how I've seen this kind of thing done in say NETCDF or PDB uber plugins. 

Doing it here in the CommonPluginInfo routines might be better, especially for `ST` type databases that would otherwise wind up doing that kind of work for every time state in the database being opened.

One thing. Must you read the *whole* xml file here? I mean, can you get away with reading just a *header* or some number of initial lines? Not a big issue but this is I/O work VisIt will do to simply decide if this plugin is the right plugin to use on the file too. So, if the XML file is *huge* and this plugin is *ahead* of others it might try when guessing the plugin from file extension, that could reprsent a lot of work as part of VisIt's file extension-to-plugin guessing process.

Reply:
The only plugins supporting .vtm and .pvd is the VTK plugin, so I think we are okay there.
Could possibly change how the information from the .xml files is garnered, but would have to use something other than vtkXMLDataParser.  If you think this could be problematic, I can look into other ways.

Reply:
Ok, so I would NOT worry about this at the moment. However, it might be good to add a teeny bit of logic there that generates a message on `debug1` when the XML file is larger than maybe the original design here had anticipated. So, maybe `stat` the file before reading it or if the XML lib you are using gives you a way to ask about file's size in bytes and then maybe issue a remark to `debug1` if file size is substantially larger than original design assumes. Maybe set the threshold at 100Kb or 1Mb. 

I dunno...maybe I am being too speculative. But, the database plugin sequence is complicated and can fail or bog down for a whole slew of subtle reasons.